### DISCUSSION TITLE
Upgrade from v2.1.2 to v2.6 producing 1051 new SQL DQD errors

### DISCUSSION THREAD

**Discussion by Ram-SN** [OHDSI/DataQualityDashboard#548] [Q&A] [answered]
Hi,

As mentioned in a previous Q&A post, we have successfully updated the application from version v2.1.2 to v2.6. When the dqd_script.r was triggered we were notified of 1051 new SQL errors that we didnt previously see in the version 2.1.2. We noticed two different kinds of errors: 
1. Unrecognized name:na at [19:70] (Level: Concept, Table
: Measurement, Field: Measurement_concept_ID, Check: plausibleUnitConceptIds)
2. Failed to parse input string "1950-06-01" (Check: plausibleTemporalAfter)

![unrec_na_error](https://github.com/OHDSI/DataQualityDashboard/assets/56275846/dd083c54-1cb9-4769-b8b6-75603c79ba75)
![date_parse_error](https://github.com/OHDSI/DataQualityDashboard/assets/56275846/db577bbf-942e-4979-8d7f-a8860e245985)

Both of these seem to be parsing errors which did not arise in the previous version. We have not changed the source of tables (GBQ) when uprading to new version.

Any help/input into this matter is greatly appreciated, thanks!

**Comment by katy-sadowski** [comment]
Thanks for the report!  

I see what the issue is in plausibleTemporalAfter and have filed this ticket to fix the bug: https://github.com/OHDSI/DataQualityDashboard/issues/549.

For plausibleUnitConceptIds - could you please share the full error report that shows the faulty SQL query?

Thank you :)

**Reply by Ram-SN** [reply]
Hi,

Attaching the expanded row from the DQD here
![date_parse_full_erros_2](https://github.com/OHDSI/DataQualityDashboard/assets/56275846/cbddc03d-2e82-4eed-8aaa-7c0c713312a9)
![date_parse_full_erros](https://github.com/OHDSI/DataQualityDashboard/assets/56275846/4cb5a926-c0df-40d8-ad5a-8e436bb7aceb)

**Reply by katy-sadowski** [reply]
Hm, this looks like it's because there are no plausible units listed for that concept ID.  I looked and I don't see that concept ID in the default concept level threshold file. Are you using a modified version of the file?

**Reply by Ram-SN** [reply]
Yes we are using a slightly modified version, so do you suggest we edit this concept_id in the concept level csv?

**Reply by katy-sadowski** [reply]
Yes - I'd make sure that if the plausibleUnitConceptIdsThreshold is populated, there are plausibleUnitConceptIds listed.  I would also recommend following the lists of plausibleUnitConceptIds from the latest release as this check was overhauled for accuracy recently.  Let me know if that works :)

**Reply by Ram-SN** [reply]
Hi, Im trying to downgrade to a previous version but the script to executeDQ seems to still be picking the plausibleTemporalAfter SQL query from v2.6. Made sure the package version is what i want.Is there something Im missing out?

**Reply by katy-sadowski** [reply, marked as answer]
It looks like this bug has been present since version 1.4! Maybe you previously had this check disabled?

**Reply by Ram-SN** [reply]
Yes we had to downgrade an even older version due to version mismatch on our end. Looking forward to the patch in the newest version to fix the plausibleTemporalAfter bug. When can we expect this? Thank you for the support!

**Reply by katy-sadowski** [reply]
I'm hoping to ship the fix with the next release in the next couple of weeks :)

**Reply by katy-sadowski** [reply]
The fix is merged into develop: https://github.com/OHDSI/DataQualityDashboard/issues/549

It'll be another week or so before this is released in a new version.  I'll keep you posted!

**Reply by Ram-SN** [reply]
Hi Katy, following up on this, thanks!

**Reply by katy-sadowski** [reply]
The release will probably go out next week. I'm completing my final task for the release currently (lots of new documentation!) and then that'll need to be reviewed.

One thing I realized is that if you're planning to upgrade to the latest DQD version then you can/should actually just use the newer versions of this check - plausibleAfterBirth,  plausibleBeforeDeath, and plausibleStartBeforeEnd - and disable plausibleTemporalAfter and plausibleDuringLife. For more info on that see our last release notes: https://ohdsi.github.io/DataQualityDashboard/news/index.html#new-checks-2-6-0.

Either way we're fixing the bug, but plausibleTemporalAfter is actually deprecated in favor of those new ones :)

**Reply by katy-sadowski** [reply]
This should be fixed in the v2.6.1 release. Thanks for your patience!
